docs/docs/starlark-reference/directory.md

```diff
@@ -0,0 +1,36 @@
+---
+title: Directory
+sidebar_label: Directory
+---
+
+The `Directory` constructor creates a `Directory` object that represents a directory inside an existing service (see
+the [ServiceConfig][service-config] object).
+Directory object can be either a files artifact or a persistent directory, depending on the arguments passed to the 
+constructor.
+
+```python
+file_artifact_directory = Directory(
+    artifact_name=files_artifact_1,
+)
+# Or:
+persistent_directory = Directory(
+    persistent_key="data-directory"
+)
+```
+
+A directory composed of a files artifact will be automatically provisioned with the given files artifact content and 
+will be immutable. The service on which the directory is mounted will not be able to modify the
```

Comment:
Isn't this also untrue? Once the files artifact's contents are copied in, the service can modify freely

Reply:
Really? I didn't know that. Isn't this an anti-pattern though? Why would a service modify a files artifact? And also, the modification would not be propagated to the artifact stored in the APIC, so it's pretty confusing. I thought we blocked this, and if not shouldn't we block it?

Reply:
I'm not actually sure how we'd block it - on k8s we could probably do it by mounting the volume as ReadOnly, but on Docker I don't think we can.

Reply:
I'm pretty sure we can mount a volume in read only mode in Docker.
I just tried
```
docker run --name test-volume --rm -v test:/data:ro **docker/getting-started**
```
(notice ti `:ro` at the end of the volume)

and that volume is read only from the container perspective:
```
/ # cd /data/
/data # cat test.txt 
Hello world!
/data # echo 'Bonjour' >> test.txt 
/bin/sh: can't create test.txt: Read-only file system
/data # 
```

Reply:
I'll resolve this comment to be able to merge, but let me know if you'd like to keep track of this, think more about it, of just action it. IDT it would be too complex to put in place, but that's maybe not something we want to do, deferring to you on this. ping me directly if you want to talk more about it